Mantle/MTLModel+Difference.m

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 Lightricks. All rights reserved.
+// Created by Maxim Grabarnik.
+
+#import "MTLModel+Difference.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation MTLModel (Difference)
+
+- (NSArray<NSString *> *)differenceFrom:(id)object {
+  MTLModel *actual = (MTLModel *)object;
+
+  if(![object isKindOfClass:self.class]) {
+		NSString *reason = [NSString stringWithFormat:@"Expected object of type %@, got : %@",
+																									self.class, [object class]];
+		[[[NSException alloc] initWithName:NSInvalidArgumentException	reason:reason
```

Comment:
is this is an assert, I think we can use NSAssert, no? seems like this will happen only in bugs.

Reply:
replaced with assert and added test for mismatched type.